docs/composite_image_map_type.js

```diff
@@ -0,0 +1,266 @@
+    // signaled by #releaseTile below.
+    const abortController = new AbortController();
+    const signal = abortController.signal;
+    this.maybeSendEvent(true);
+    this.tileMap.set(tileDiv, abortController);
+    let remainingTiles = tileUrls.length;
+    // Process each image as it comes in, so user can see latest images without
+    // waiting for slow-loading ones.
+    const promiseProcessor = (url) => {
+      // If releaseTile() was called for this tile before this promise
+      // completed, we are aborting.
+      if (!this.tileMap.get(tileDiv)) {
+        if (typeof (url) === 'string') {
+          // If the image was already downloaded, free it.
+          URL.revokeObjectURL(url);
+        }
+        return;
+      }
+      if (typeof (url) === 'string') {
+        this.appendImage(url, tileDiv);
+      } else if (url instanceof ErrorObject) {
+        if (url.statusCode !== NOT_FOUND_STATUS_CODE) {
+          console.error(url.statusCode, url.getMessage());
+        }
+      }
+      if (!--remainingTiles) {
+        // All fetches have completed, maybe some with errors. Tile is done.
+        this.markTileCompleted(tileDiv);
+      }
+    };
+    tileUrls.map((url) => fetchWithBackoff(url, signal).then(promiseProcessor));
+    return tileDiv;
+  }
+
+  /**
+   * Implements MapType#releaseTile. Revokes local object URL for all images in
+   * the given div, freeing up memory, and canceling any in-flight fetches.
+   * @param {HTMLDivElement} div
+   */
+  releaseTile(div) {
+    const controller = this.tileMap.get(div);
+    if (controller) {
+      controller.abort();
+      this.markTileCompleted(div);
+    }
+  }
+
+  /**
+   * Creates and appends an HTMLImageElement pointing at the given url. When the
+   * image is loaded, the url (which points at a local object) will be freed to
+   * save memory.
+   * @param {string} url URL from {@link URL#createObjectURL}
+   * @param {HTMLDivElement} tileDiv Div to attach image to.
+   */
+  appendImage(url, tileDiv) {
+    const img = tileDiv.ownerDocument.createElement('img');
+    img.src = url;
+    img.opacity = this.opacity;
+    // Stack images over each other.
+    img.style.position = 'absolute';
+    // Clear blob from memory as soon as it is loaded.
+    img.onload = () => URL.revokeObjectURL(url);
+    tileDiv.appendChild(img);
+  }
+
+  /**
+   * Removes div from {@link this#tileMap} and sends an event if no tiles remain
+   * to be rendered.
+   * @param {HTMLDivElement} div
+   */
+  markTileCompleted(div) {
+    this.tileMap.delete(div);
+    this.maybeSendEvent(false);
+  }
+
+  /**
+   * If we are transitioning from no pending fetches to pending fetches or vice
+   * versa, send an event.
+   * @param {boolean} adding True if we are about to start a fetch, false if we
+   *     have just finished all fetches
+   */
+  maybeSendEvent(adding) {
+    if (!this.tileMap.size) {
+      const event = new Event('tile');
+      event.count = adding ? 1 : 0;
```

Comment:
Technically it could, but I'm trying to keep the "TileEvent" interface as similar as possible to ee.MapLayerOverlay, which has a number in this field. If I put a boolean in, it would fail to be === 0, which is what layer_util checks for.

As it stands, this event is always telling the truth: when we fire, there really are 0 outstanding tiles or 1 outstanding tiles. We just choose to suppress the other events: counts >1, and count === 1 when decreasing.